File: src/test/java/codebreaker/CodeBreakerSteps.java
package codebreaker;

import cuke4duke.*;
import static org.junit.Assert.*;

import java.util.regex.Pattern;

public class CodeBreakerSteps {
    private Game game;
    private String mark;
    
    @Given("the secret code is (.*)")
    public void theSecretCodeIs(String code) {
        game = new Game(stripSpace(code));
    }

    @When("I guess (.*)")
    public void iGuess(String code) {
        mark = game.guess(stripSpace(code));
    }

    @Then("the mark should be (.*)")
    public void theMarkShouldBe(String code) {
        assertEquals(code, mark);
    }

    private String stripSpace(String s) {
       return s.replaceAll(" ", "");        
    }
}

File: src/test/java/codebreaker/code_breaker_steps.rb
require 'spec/expectations'
include_class 'codebreaker.Game'

Given(/the secret code is (.*)/) do |code|
    @game = Game.new(strip_space(code))
end

When(/I guess (.*)/) do |code|
    @mark = @game.guess(strip_space(code))
end

Then(/the mark should be (.*)/) do |code|
    @mark.should == code
end

def strip_space(s)
   s.gsub(" ", "")   
end
